torch/linalg/__init__.py

```diff
@@ -140,6 +140,73 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+cond = _add_docstr(_linalg.linalg_cond, r"""
+linalg.norm(input, p=None, *, out=None) -> Tensor
+
+Computes the condition number of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`, using the matrix norm defined by :attr:`p`.
+The condition number is defined as the matrix norm of
+:attr:`input` times the matrix norm of the inverse of :attr:`input`.
+
+This function supports real-valued, and only on CPU, complex-valued input.
+
+.. note:: For `p = {None, 2, -2}` :attr:`input` can be non-square. For other norm types :attr:`input` must be
+          a square matrix or a batch of square matrices. If :attr:`input` does not satisfy the requirement
+          then a RuntimeError will be thrown.
+
+.. note:: For the batched input if at least one matrix in the batch is not invertible,
+          currently getting the result for all other (possibly) invertible matrices in the batch is not implemented.
+
+Args:
+    input (Tensor): the input matrix of size :math:`(m, n)` or the batch of matrices of size :math:`(*, m, n)`
+                    where `*` is one or more batch dimensions.
+
+    p (int, float, inf, -inf, 'fro', 'nuc', optional): the type of the matrix norm to use in the computations.
+        The following norms are supported:
+
+        =====  ============================
+        p      norm for matrices
+        =====  ============================
+        None   2-norm (ratio of the largest singular value to the smallest singular value)
+        'fro'  Frobenius norm
+        'nuc'  nuclear norm
+        inf    max(sum(abs(x), dim=1))
+        -inf   min(sum(abs(x), dim=1))
+        1      max(sum(abs(x), dim=0))
+        -1     min(sum(abs(x), dim=0))
+        2      2-norm (ratio of the largest singular value to the smallest singular value)
+        -2     ratio of the smallest singular value to the largest singular value
+        =====  ============================
+
+        Default: ``None``
+
+Keyword args:
+    out (Tensor, optional): The output tensor. Ignored if ``None``. Default: ``None``
+
+Examples::
+
+    >>> from torch import linalg as LA
+    >>> a = torch.tensor([[1., 0, -1], [0, 1, 0], [1, 0, 1]])
```

Comment:
This is a good example but let's add a batched example, too

Reply:
I added a batched example, and one more with a complex dtype. I didn't repeat all the norm types.
```python
    >>> a = torch.randn(3, 4, 4)
    >>> LA.cond(a)
    tensor([ 4.4739, 76.5234, 10.8409])

    >>> a = torch.randn(3, 4, 4, dtype=torch.complex64)
    >>> LA.cond(a)
    tensor([ 5.9175, 48.4590,  5.6443])
    >>> LA.cond(a, 1)
    >>> tensor([ 11.6734+0.j, 105.1037+0.j,  10.1978+0.j])
```